packages/arb-provider-ethers/src/lib/provider.ts

```diff
@@ -256,15 +220,25 @@ export class ArbProvider extends ethers.providers.BaseProvider {
     return null
   }
 
-  public async getMessageResult(
-    ethTxHash: string
-  ): Promise<MessageResult | null> {
-    const ethReceipt = await this.ethProvider.waitForTransaction(ethTxHash)
-    const arbTxId = await this.getArbTxId(ethReceipt)
-    if (!arbTxId) {
-      return null
+  public async getMessageResult(txHash: string): Promise<MessageResult | null> {
+    // Transaction hash could be either an L1 transaction or
+    // an arbitrum transaction
```

Comment:
Added

Reply:
The comments don't address the potential security issue I was trying to point out.  If someone can make an Eth Tx and a separate Arb Tx that both have the same hash, then this would cause this code to have unexpected and possibly risky behavior. So we need to explain why it's not possible to construct an Eth Tx and an Arb Tx that have the same hash. (Or if that is possible, to change how Arb Tx's are hashed, to make it impossible.)

Reply:
I'm not sure if you saw the latest version of the comment since the above box is showing outdated code. Currently it reads:
```
// Transaction hash could be either an L1 transaction or
// an arbitrum transaction. An L1 transaction hash is over a very different
// data structure compared to the L2 transaction hash and so no collisions
// are possible since both use keccak256 which is cryptographically secure
```

Happy to update it if you have any suggestions

Reply:
Comment updated and issue created as discussed.